Documentation/design-docs/COM-activation.md

```diff
@@ -0,0 +1,105 @@
+# COM Activation for .NET Core on Windows
+
+## Purpose
+
+In order to more fully support the vast number of existing .NET Framework in their transition to .NET Core, support of the COM Activation scenario in .NET Core is required. Without this support it is not possible for many .NET Framework consumers to even consider transitioning to .NET Core. The intent of this document is to describe aspects of the activation of a .NET class written for .NET Core within a COM environment. This support includes but is not limited to activation via the [`CoCreateInstance()`](https://docs.microsoft.com/en-us/windows/desktop/api/combaseapi/nf-combaseapi-cocreateinstance)API in C/C++ or from a [Windows Script Host](https://docs.microsoft.com/en-us/windows/desktop/com/using-com-objects-in-windows-script-host).
+
+### Requirements
+
+* Discover all installed versions of .NET Core.
+* Load the appropriate version of .NET Core for the class.
+* Instantiate an instance of the class and return a pointer to an [`IUnknown`](https://docs.microsoft.com/en-us/windows/desktop/api/unknwn/nn-unknwn-iunknown) that represents the newly constructed class.
+* Ensure that if an existing CLR instance of the correct version is already present in process, that existing one is used.
```

Comment:
Do we know if there's an existing mechanism to query for loaded CLR instances in the process (and more importantly have a way to talk to them in a safe way)? I'm mostly just curious as this feels like a rather tricky bit.

Reply:
That is a very tricky bit and something I am looking for feedback on. I think the .NET Framework's hosting API handles that logic, but I do not yet know if this support is within the .NET Core code base. @jkotas or @davidwrighton thoughts?

Reply:
Yes, .NET Framework hosting APIs have logic for this. There is nothing like that in .NET Core.

Loading multiple runtimes into a process was possible in theory in .NET Framework. In practice,  most people tried to stay away from it. 

I think we may want to limit .NET Core 3.0 support to having one runtime in a process. I do not think it is worth  opening the multiple runtimes rat-hole just for COM interop.

Reply:
Sounds good. I will update the requirements.

Reply:
Actually I am bit confused here. If we aren't going to allow multiple instances, then we have the unfortunate situation where the first COM activation sets the precedent for all further activations? I think this could be relaxed to, only 1 version of each runtime will be created from the loaded version of `hostfxr`?